code/__DEFINES/milla.dm

```diff
@@ -0,0 +1,129 @@
+		// Then check in the current directory.
+		if(fexists("./libmilla.so"))
+			return __milla = "./libmilla.so"
+		// And elsewhere.
+		return __milla = "libmilla.so"
+	else
+		// First check if it's built in the usual place.
+		if(fexists("./milla/target/i686-pc-windows-msvc/release/milla.dll"))
+			return __milla = "./milla/target/i686-pc-windows-msvc/release/milla.dll"
+		// Then check in the current directory.
+		if(fexists("./milla.dll"))
+			return __milla = "./milla.dll"
+		// And elsewhere.
+		return __milla = "milla.dll"
+
+#define MILLA (__milla || __detect_milla())
+
+#define MILLA_CALL(func, args...) call_ext(MILLA, "byond:[#func]_ffi")(args)
+
+#define REAL_SET_TILE_ATMOS(args...) MILLA_CALL(set_tile_atmos, args)
+
+/proc/set_tile_atmos(x, y, z, blocked_north, blocked_east, blocked_south, blocked_west, atmos_mode, external_temperature, oxygen, carbon_dioxide, nitrogen, toxins, sleeping_agent, agent_b, temperature, innate_heat_capacity)
+	if(!isnum(x) || !isnum(y) || !isnum(z))
+		CRASH("Called set_tile_atmos without specifying coordinates.")
+	if(!isnull(blocked_north) && !isnum(blocked_north))
+		CRASH("Called set_tile_atmos with non-number blocked_north.")
+	if(!isnull(blocked_east) && !isnum(blocked_east))
+		CRASH("Called set_tile_atmos with non-number blocked_east.")
+	if(!isnull(blocked_south) && !isnum(blocked_south))
+		CRASH("Called set_tile_atmos with non-number blocked_south.")
+	if(!isnull(blocked_west) && !isnum(blocked_west))
+		CRASH("Called set_tile_atmos with non-number blocked_west.")
+	if(!isnull(atmos_mode) && !isnum(atmos_mode))
+		CRASH("Called set_tile_atmos with non-number atmos_mode.")
+	if(!isnull(external_temperature) && !isnum(external_temperature))
+		CRASH("Called set_tile_atmos with non-number external_temperature.")
+	if(!isnull(oxygen) && !isnum(oxygen))
+		CRASH("Called set_tile_atmos with non-number oxygen.")
+	if(!isnull(carbon_dioxide) && !isnum(carbon_dioxide))
+		CRASH("Called set_tile_atmos with non-number carbon_dioxide.")
+	if(!isnull(nitrogen) && !isnum(nitrogen))
+		CRASH("Called set_tile_atmos with non-number nitrogen.")
+	if(!isnull(toxins) && !isnum(toxins))
+		CRASH("Called set_tile_atmos with non-number toxins.")
+	if(!isnull(sleeping_agent) && !isnum(sleeping_agent))
+		CRASH("Called set_tile_atmos with non-number sleeping_agent.")
+	if(!isnull(agent_b) && !isnum(agent_b))
+		CRASH("Called set_tile_atmos with non-number agent_b.")
+	if(!isnull(temperature) && !isnum(temperature))
+		CRASH("Called set_tile_atmos with non-number temperature.")
+	if(!isnull(innate_heat_capacity) && !isnum(innate_heat_capacity))
+		CRASH("Called set_tile_atmos with non-number innate_heat_capacity.")
+	return REAL_SET_TILE_ATMOS(x, y, z, blocked_north, blocked_east, blocked_south, blocked_west, atmos_mode, external_temperature, oxygen, carbon_dioxide, nitrogen, toxins, sleeping_agent, agent_b, temperature, innate_heat_capacity)
+
+#define REAL_GET_TILE_ATMOS(args...) MILLA_CALL(get_tile_atmos, args)
```

Comment:
Dunno about the `REAL_*` prefix on these, unless these were to get undefined in this file

Reply:
Yeah, all of the defines here should get undef'd, good point. The procs are the API that everything else should use.

Reply:
Done. One does wonder if there's a better folder for the file now, though, since it doesn't actually expose any defines.